manager/controllers/app/m4dapplication_controller.go

```diff
@@ -120,20 +123,38 @@ func (r *M4DApplicationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 	return ctrl.Result{}, nil
 }
 
-func checkResourceStatus(applicationContext *app.M4DApplication, status app.ObservedState) {
+func (r *M4DApplicationReconciler) checkReadiness(applicationContext *app.M4DApplication, status app.ObservedState) error {
```

Comment:
Just to be sure: the logic here only runs once after success right?

Reply:
checkReadiness runs on every Plotter update. The post-processing (registering data assets) runs once.